modules/disk.sh

```diff
@@ -96,7 +96,7 @@ disk_align_sectors() {
 # $2 - destination file
 # $3 - offset in number of 512 sectors
 disk_write_at_offset() {
-  run_and_log_cmd "dd if=${1} of=${2} seek=${3} conv=sparse status=none"
+  run_and_log_cmd "dd if=${1} of=${2} seek=${3} conv=sparse,notrunc status=none"
```

Comment:
What is the motivation behind this change?

Reply:
With GPT there should be a backup partition table at the end of the disk.  So before this change, `mender-convert-package` creates the output image file, writes the partition table, and when it writes the first partition data the file is truncated and the backup partition table is gone.

I don't think this change causes any worse performance, it leaves the backup partition table, and I think it makes troubleshooting easier.

An alternative would be to run `sgdisk -e` on the file after it's done (I haven't tested this on the image but it should work).  As a side note, if the output image is written to a disk that is larger than the image, the backup partition table will end up in the wrong place.  This can be fixed by running `sgdisk -e` and I have tested it.  The device will still boot without the backup partition table so I'm not sure if this important or not.  It might be good to either document this or make it automatic as part of any "first boot data partition resize" operations.  It doesn't seem like my images are resizing the data partition on first boot though, so I'm thinking maybe that is just for Yocto images?

Reply:
I see now that the GROWFS option only works with systemd 242 and newer, so that's why it wasn't working for me (my test system has version 241).  Since that's part of systemd there's not really any script to tap into for this purpose.

I'm not sure if it would be better to just document this as an optional extra step when writing the image to the device, or if there's a good way to run the `sgdisk -e` command on first boot. Any suggestions?

Reply:
> I don't think this change causes any worse performance, it leaves the backup partition table, and I think it makes troubleshooting easier.

Makes sense, could you put this in a separate commit as well. I think your comments would be a good commit message

Reply:
> I see now that the GROWFS option only works with systemd 242 and newer

There was a discussions regarding this on MH today,

https://hub.mender.io/t/systemd-growfs-resize-to-128m-only/1446/7